src/lightning/data/cache/cache.py

```diff
@@ -66,19 +68,23 @@ def __setitem__(self, index: int, data: Any) -> None:
         """Store an item in the writer."""
         self._writer[index] = data
 
+    def _add_item(self, index: int, data: Any) -> Optional[str]:
+        """Store an item in the writer and optionally return the chunk path."""
+        return self._writer.add_item(index, data)
+
     def __getitem__(self, index: Union[int, ChunkedIndex]) -> Dict[str, Any]:
         """Read an item in the reader."""
         if isinstance(index, int):
             index = ChunkedIndex(index, self._get_chunk_index_from_index(index))
         return self._reader.read(index)
 
-    def done(self) -> None:
+    def done(self) -> Optional[List[str]]:
```

Comment:
I'd expect a simple boolean as return type here.

Reply:
This is required to ensure things fly properly.